### ISSUE TITLE
python?

### ISSUE THREAD

**Issue by argriffing** [weijianzhang/MatrixDepot.jl#6]
This might be a dumb question, but is there a Python interface?  This possibility was mentioned in https://media.readthedocs.org/pdf/matrix-repository/latest/matrix-repository.pdf which seems to have been merged into MatrixDepot.jl.

**Comment by weijianzhang** [comment]
Hi @argriffing,  thanks for the question. But unfortunately, I haven't found time to implement it. 
I think I will work on it this summer.

**Comment by argriffing** [comment]
Even if this is not implemented, would you know the best way to get access to matrix sets 1--4 mentioned in section 6 of http://core.ac.uk/download/pdf/17664.pdf in a way that could be included in MIT or BSD licensed code (not GPL)?  I've noticed a typo in my implementation of that expm code which wasn't caught by less comprehensive testing.

**Comment by weijianzhang** [comment]
Hi, 

You may want to try rogues (https://pypi.python.org/pypi/rogues/0.2.0), which is MIT licensed. 

Best wishes,

Weijian

**Comment by argriffing** [comment]
Thanks for the link! It does say that it's MIT licensed, but its sources appear to be incompatibly licensed (TOMS-licensed and GPL-licensed). @macd?

**Comment by macd** [comment]
Well, that is a tricky question.  I did these awhile ago when I was first learning numpy / scipy.   As it says in the readme, I mostly used version 3.0, which was downloaded from Prof. Higham website.  I didn't see how that version was licensed. So that is unclear to me.  I didn't use any of the 1989 TOMS code, but I did look at it.

So, the situation is murky, at best.  I know some folks have a hard time with the older TOMS
license (the newer version seems OK?).  I don't believe there is anything here that is GPL'd
however. @weijianzhang how is the Julia version licensed exactly?  You must have used the old MATLAB implementations from Prof. Higham?  You are in Prof. HIgham's research group?  I see you have an MIT "expat" license.  Anything special done for that licensing?

**Comment by argriffing** [comment]
> As it says in the readme, I mostly used version 3.0, which was downloaded from Prof. Higham website.
[...]
> I don't believe there is anything here that is GPL'd however.

The [website](http://www.maths.manchester.ac.uk/~higham/mctoolbox/) says that both the Matrix Computation Toolbox and the Test Matrix Toolbox are distributed under the terms of the [GNU General Public License](http://www.gnu.org/copyleft/gpl.html) (version 3 of the License, or any later version) as published by the Free Software Foundation.  But this doesn't necessarily mean that they are not also distributed under other licenses. Separately, I'm not sure about possible TOMS-related issues.

**Comment by macd** [comment]
Ah, well, I haven't looked at that website in awhile.  I certainly don't
remember
that to be the case in 2008-2009 time frame and none of the MATLAB code
includes a license.  But that is not to say that it wasn't GPL'd then, only
I
don't remember.

**Comment by weijianzhang** [comment]
Hi @macd,

>how is the Julia version licensed exactly? You must have used the old MATLAB implementations from Prof. Higham? You are in Prof. HIgham's research group? I see you have an MIT "expat" license. Anything special done for that licensing?

MatrixDepot is distributed under MIT "expat" license. I did look at Higham's MATLAB implementations of many matrix generators.  But the Julia implementations are quite different and I coded the test matrices based on the original papers. Also in Julia, loops are fast and my implementations use loops instead of the vectorized expressions of MATLAB. 

Yes, I am in Prof. Nick Higham's research group.

Best wishes,

Weijian
